modules/apps/headless/headless-builder/headless-builder-api/src/main/java/com/liferay/headless/builder/application/provider/APIApplicationProvider.java

```diff
@@ -21,7 +21,7 @@
  */
 public interface APIApplicationProvider {
 
-	public APIApplication getAPIApplication(String baseURL, long companyId)
+	public APIApplication fetchAPIApplication(String baseURL, long companyId)
```

Comment:
If we transform into a `fetch` method, shouldn't we avoid throwing an Exception?
AFAIK, it is the pattern that we follow in Liferay (at least in service builder)

Reply:
Yes, in service builder it is like that, but not in other parts of Liferay. I considered throwing an exception when it is an unexpected exception. I don't think it is a good idea to return null if there is a server error or any other kind of unexpected error. Maybe better to expand the exception to make the components fail in the original invocation. 

What do you think about that?